PR for post filter outputs endpoint

j-s-rawat opened the pull request:
### What

[Describe what you have changed and why.]

https://trello.com/c/yC28daUV/5468-create-post-filter-outputs-endpoint

### How to review

The changes support POST endpoint for 'filter-outputs'
A UT added for validation, and currently the only validation check is to make sure that input body has non-empty data

### Who can review

Any back-end dev that has previously worked on endpoints

PauloPortugal commented on api/api.go:
```diff
@@ -50,8 +50,10 @@ func New(ctx context.Context, cfg *config.Config, r chi.Router, idc *identity.Cl
 
 func (api *API) enablePublicEndpoints() {
 	api.Router.Post("/filters", api.createFilter)
+	api.Router.Post("/filter-outputs", api.createFilterOutput)
```
```suggestion
        api.Router.Post("/filter-outputs", api.createFilterOutput)
	api.Router.Post("/filters", api.createFilter)
```

PauloPortugal replied:
I would swap the order of these in order to keep the `/filter` resources together.
Improves readability

PauloPortugal commented on api/api.go:
```diff
@@ -63,6 +65,7 @@ func (api *API) enablePrivateEndpoints() {
 	r.Use(permissions.Require(auth.Permissions{Read: true}))
 
 	r.Post("/filters", api.createFilter)
+	r.Post("/filter-outputs", api.createFilterOutput)
```
```suggestion
	r.Post("/filter-outputs", api.createFilterOutput)
	r.Post("/filters", api.createFilter)
```

PauloPortugal commented on api/contract.go:
```diff
@@ -44,6 +45,23 @@ type createFilterResponse struct {
+	Downloads model.FilterOutput `bson:"downloads"      json:"downloads"`
+}
+
+func (r *createFilterOutputsRequest) Valid() error {
+
+	return nil
+}
```
Do we need to introduce some logic to validate the `POST` request?

j-s-rawat replied:
A very basic validation has been addressed in validateFilterOutput

PauloPortugal reviewed (changes requested):
Please review the comments.
On a side note, I would also advise to [rebase & fixup](https://www.mikulskibartosz.name/git-fixup-explained/) your commit history as you have 7 commits that could well be bundled into one or two.
This is a non-blocker, as I just like good git hygiene.

j-s-rawat commented on features/filter_outputs.feature:
```diff
@@ -0,0 +1,59 @@
+        {
+            "xls": 
+            {
+                "href": "string",
+                "size": "string",
+                "public": "string",
+                "private": "string",
+                "skipped": true
```
Good suggestion. The current development is done with a though that what ever is sent 'in' is added and the same is sent 'out' in response.

j-s-rawat commented on features/filter_outputs_authorised.feature:
```diff
@@ -0,0 +1,171 @@
+        {
+            "xls": 
+            {
+                "href": "string",
+                "size": "string",
+                "public": "string",
+                "private": "string",
+                "skipped": true
```
@franmoore05 @ChrysmOre, do we have some logical values for it. It won't change the test but might help in inderstanding. I doubt that there will be further changes to this piece of code as we discover furhter validation required for input

j-s-rawat commented:
> Please review the comments. On a side note, I would also advise to [rebase & fixup](https://www.mikulskibartosz.name/git-fixup-explained/) your commit history as you have 7 commits that could well be bundled into one or two. This is a non-blocker, as I just like good git hygiene.

Thanks, I will be addressing this once I have addressed all the comments/suggestions as a final step before merging to 'develop'
